File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/datasync/membership/RemoveMembership.java
package com.pubnub.api.java.endpoints.datasync.membership;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.java.models.consumer.datasync.membership.PNDataSyncRemoveMembershipResult;
import org.jetbrains.annotations.Nullable;

/**
 * @see com.pubnub.api.java.datasync.DataSync#removeMembership(String)
 */
public interface RemoveMembership extends Endpoint<PNDataSyncRemoveMembershipResult> {
    /**
     * Optional eTag for a conditional delete ({@code If-Match} header).
     */
    RemoveMembership ifMatch(@Nullable String ifMatch);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/datasync/membership/RemoveMembership.kt
package com.pubnub.api.endpoints.datasync.membership

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.datasync.membership.PNDataSyncRemoveMembershipResult

/**
 * @see [com.pubnub.api.datasync.DataSync.removeMembership]
 */
actual interface RemoveMembership : Endpoint<PNDataSyncRemoveMembershipResult>
